tests/libjlm/tooling/TestJlcCommandLineParser.cpp

```diff
@@ -122,13 +123,39 @@ Test4()
   assert(compilation.OutputFile() == "foobar.o");
 }
 
+static void
+Test5()
+{
+  /*
+   * Arrange
+   */
+  std::vector<std::string> commandLineArguments({"jlc", "foobar.c", "-Jcne", "-Jdne"});
+
+  /*
+   * Act
+   */
+  auto & commandLineOptions = ParseCommandLineArguments(commandLineArguments);
+  auto commandGraph = jlm::JlcCommandGraphGenerator::Generate(commandLineOptions);
+
+  /*
+   * Assert
+   */
+  auto & clangCommandNode = (*commandGraph->GetEntryNode().OutgoingEdges().begin()).GetSink();
+  auto & jlmOptCommandNode = (clangCommandNode.OutgoingEdges().begin())->GetSink();
+  auto command = dynamic_cast<const jlm::JlmOptCommand*>(&jlmOptCommandNode.GetCommand());
+  assert(command->Optimizations_.size() == 2);
+  assert(command->Optimizations_[0] == jlm::JlmOptCommand::Optimization::CommonNodeElimination && \
+         command->Optimizations_[1] == jlm::JlmOptCommand::Optimization::DeadNodeElimination);
+}
+
```

Comment:
The file is called TestJlcCommandLineParser.cpp and not TestJlcCommandGraphGenerator.cpp. Please add tests for the command graph generation in the appropriate file (or create one if none exists).

Reply:
It tests command line parsing, but to really verify it, one need to check the graph that's generated.
What name do you want for something that checks command line arguments that generate correct graphs.

Reply:
TestJlcCommandLineParser.cpp -> Tests command line parsing. The tests take argc and argvs as input and have the CommandLineOptions as output (that is verified).

TestJlcCommandGraphGenerator -> Tests command graph generation. The tests take a CommandLineOptions as input and have the command graph as output (that is verified).

Your added test currently tests both in a file that is for command line parser tests.

Reply:
Fixed